lib/node_modules/@stdlib/math/base/special/cinvf/test/test.native.js

```diff
@@ -0,0 +1,455 @@
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+/* eslint-disable id-length */
+
+'use strict';
+
+// MODULES //
+
+var resolve = require( 'path' ).resolve;
+var tape = require( 'tape' );
+var isnanf = require( '@stdlib/math/base/assert/is-nanf' );
+var absf = require( '@stdlib/math/base/special/absf' );
+var EPS = require( '@stdlib/constants/float32/eps' );
+var FLOAT32_SMALLEST_NORMAL = require( '@stdlib/constants/float32/smallest-normal' );
+var Complex64 = require( '@stdlib/complex/float32/ctor' );
+var real = require( '@stdlib/complex/float32/real' );
+var imag = require( '@stdlib/complex/float32/imag' );
+var tryRequire = require( '@stdlib/utils/try-require' );
+
+
+// VARIABLES //
+
+var cinvf = tryRequire( resolve( __dirname, './../lib/native.js' ) );
+var opts = {
+	'skip': ( cinvf instanceof Error )
+};
+
+
+// FIXTURES //
+
+var data = require( './fixtures/julia/data.json' );
+var largeNegativeImaginaryComponents = require( './fixtures/julia/large_negative_imaginary_components.json' );
+var largeNegativeRealComponents = require( './fixtures/julia/large_negative_real_components.json' );
+var largePositiveImaginaryComponents = require( './fixtures/julia/large_positive_imaginary_components.json' );
+var largePositiveRealComponents = require( './fixtures/julia/large_positive_real_components.json' );
+var tinyNegativeImaginaryComponents = require( './fixtures/julia/tiny_negative_imaginary_components.json' );
+var tinyNegativeRealComponents = require( './fixtures/julia/tiny_negative_real_components.json' );
+var tinyPositiveImaginaryComponents = require( './fixtures/julia/tiny_positive_imaginary_components.json' );
+var tinyPositiveRealComponents = require( './fixtures/julia/tiny_positive_real_components.json' );
+
+
+// TESTS //
+
+tape( 'main export is a function', opts, function test( t ) {
+	t.ok( true, __filename );
+	t.strictEqual( typeof cinvf, 'function', 'main export is a function' );
+	t.end();
+});
+
+tape( 'the function computes the inverse of a double-precision complex floating-point number', opts, function test( t ) {
+	var v;
+
+	v = cinvf( new Complex64( 2.0, 4.0 ) );
+	t.strictEqual( real( v ), 0.10000000149011612, 'returns expected value' );
+	t.strictEqual( imag( v ), -0.20000000298023224, 'returns expected value' );
+	t.end();
+});
+
+tape( 'the function computes a complex inverse', opts, function test( t ) {
+	var delta;
+	var qre;
+	var qim;
+	var tol;
+	var re;
+	var im;
+	var i;
+	var q;
+
+	re = data.re;
+	im = data.im;
+	qre = data.qre;
+	qim = data.qim;
+
+	for ( i = 0; i < re.length; i++ ) {
+		q = cinvf( new Complex64( re[ i ], im[ i ] ) );
+
+		if ( real( q ) === qre[ i ] ) {
+			t.strictEqual( real( q ), qre[ i ], 'returns expected real component' );
+		} else {
+			delta = absf( real( q ) - qre[ i ] );
+			tol = 5 * EPS * absf( qre[ i ] );
```

Comment:
This tolerance, and tolerances for the tests below are different from what we have in `test.js`. Is there any specific reason for this? Ideally, the tolerances much match.

Reply:
when i run the tests locally, the tolerance in `test.js` was appropriate for all test cases to pass. the c files had some more inaccuracy due to the single-precision nature of the code hence i had to increase the tolerance to pass the tests. i think this is because i used `fmaxf` instead of `stdlib_base_maxf`. will update the C code and see if i can match both the tolerances.